### PR TITLE
3.0: Remove every reference to Torque and SGE schedulers and fix some integ tests

### PR DESCRIPTION
**Pull request by enrico-usai**
`TorqueCommands` in `tests/common/schedulers_common` has been preserved because in `test_slurm` we're testing torque wrappers for slurm.

## Integ tests fixes
* Fix test tag propagation by removing substack related code
* Remove unused nodewatcher_terimates_failing_node test files
* Remove deprecate parameters from update command in test framework
* Fix SpotBidPercentage values in awsbatch update tests
* Remove unused ScaledownIdletime from test_nccl config file

### CODE REVIEW

File: tests/integration-tests/configs/develop.yaml [hunk 1]
```diff
@@ -7,19 +7,19 @@ test-suites:
   runtime_bake:
     test_runtime_bake.py::test_runtime_bake:  # These are currently skipped
       dimensions:
-        {%- for os, scheduler in [("alinux2", "slurm"), ("alinux2", "torque"), ("centos7", "sge")] %}
+        {%- for os, scheduler in [("alinux2", "slurm")] %}
```
Comment (demartinofra) [inline tests/integration-tests/configs/develop.yaml:10]: you can probably remove the for loops now. Shall we leave centos7 with slurm to test the runtime bake on that os?

File: tests/integration-tests/configs/develop.yaml [hunk 2]
```diff
@@ -7,19 +7,19 @@ test-suites:
...
         {%- endfor %}
-        {%- for os, scheduler in [("ubuntu1804", "sge")] %}
+        {%- for os, scheduler in [("ubuntu1804", "slurm")] %}
         - regions: ["us-gov-west-1"]
           instances: {{ common.INSTANCES_DEFAULT_X86 }}
           oss: ["{{ os }}"]
           schedulers: ["{{ scheduler }}"]
         {%- endfor %}
-        {%- for os, scheduler in [("ubuntu1804", "sge"), ("alinux2", "slurm")] %}
+        {%- for os, scheduler in [("alinux2", "slurm")] %}
```
Comment (demartinofra) [inline tests/integration-tests/configs/develop.yaml:22]: shall we leave ubuntu18 with arm?

File: tests/integration-tests/tests/cfn-init/test_cfn_init.py [hunk 3]
```diff
@@ -23,7 +23,7 @@
 
 @pytest.mark.regions(["eu-central-1"])
 @pytest.mark.instances(["c5.xlarge"])
-@pytest.mark.schedulers(["slurm", "sge"])
+@pytest.mark.schedulers(["slurm"])
```
Comment (demartinofra) [inline tests/integration-tests/tests/cfn-init/test_cfn_init.py:26]: probably not part of this PR but we should get rid of the regions, instances, schedules. oss annotations since now we use the config file

File: tests/integration-tests/tests/common/schedulers_common.py [hunk 4]
```diff
@@ -601,7 +499,6 @@ def get_nodes_status(self):
 def get_scheduler_commands(scheduler, remote_command_executor):
     scheduler_commands = {
         "awsbatch": AWSBatchCommands,
-        "sge": SgeCommands,
         "slurm": SlurmCommands,
         "torque": TorqueCommands,
```
Comment (demartinofra) [inline tests/integration-tests/tests/common/schedulers_common.py:503]: don't you need to remove Torque as well?

File: tests/integration-tests/tests/scaling/test_scaling/test_nodewatcher_terminates_failing_node/pcluster.config.yaml [hunk 5]
```diff
@@ -1,22 +0,0 @@
-Image:
```
Comment (demartinofra) [inline tests/integration-tests/tests/scaling/test_scaling/test_nodewatcher_terminates_failing_node/pcluster.config.yaml:1]: have you removed the test code as well?

File: tests/integration-tests/configs/develop.yaml [hunk 6]
```diff
@@ -7,19 +7,19 @@ test-suites:
   runtime_bake:
```
Comment (demartinofra) [inline tests/integration-tests/configs/develop.yaml:7]: probably you need to look at the common config as well

File: tests/integration-tests/tests/update/test_update/test_update_awsbatch/pcluster.config.update.yaml [hunk 7]
```diff
@@ -24,4 +24,4 @@ Scheduling:
           MinvCpus: 2
           MaxvCpus: 4
           DesiredvCpus: 3
-          SpotBidPercentage: 70
\ No newline at end of file
+          SpotBidPercentage: 0.7
```
Comment (demartinofra) [inline tests/integration-tests/tests/update/test_update/test_update_awsbatch/pcluster.config.update.yaml:27]: BidPercentage in Batch is an integer. why changing the type?

File: tests/integration-tests/configs/develop.yaml (hunk 1 above)
Comment (enrico-usai) [reply tests/integration-tests/configs/develop.yaml:10]: Runtime bake are currently skipped and they are no longer supported but I'll let @lukeseawalker and @yuleiwan fix them.

File: tests/integration-tests/tests/common/schedulers_common.py (hunk 4 above)
Comment (enrico-usai) [reply tests/integration-tests/tests/common/schedulers_common.py:503]: Nope, see PR description:
> TorqueCommands in tests/common/schedulers_common has been preserved because in test_slurm we're testing torque wrappers for slurm.

File: tests/integration-tests/tests/scaling/test_scaling/test_nodewatcher_terminates_failing_node/pcluster.config.yaml (hunk 5 above)
Comment (enrico-usai) [reply tests/integration-tests/tests/scaling/test_scaling/test_nodewatcher_terminates_failing_node/pcluster.config.yaml:1]: It was already removed by @hanwen-pcluste that missed to remove these files.

File: tests/integration-tests/configs/develop.yaml (hunk 2 above)
Comment (enrico-usai) [reply tests/integration-tests/configs/develop.yaml]: Restored

File: tests/integration-tests/tests/update/test_update/test_update_awsbatch/pcluster.config.update.yaml (hunk 7 above)
Comment (enrico-usai) [reply tests/integration-tests/tests/update/test_update/test_update_awsbatch/pcluster.config.update.yaml]: Good catch! Fixed. We were using float for both SpotPrice and SpotBidPercentage

File: tests/integration-tests/tests/scaling/test_scaling.py [hunk 8]
```diff
@@ -59,48 +59,27 @@ def test_multiple_jobs_submission(scheduler, region, pcluster_config_reader, clu
...
-        expected_asg_capacity=(0, 3),
-        expected_compute_nodes=(0, 3),
-    )
+    _assert_scaling_works(compute_nodes_time_series=compute_nodes_time_series, expected_compute_nodes=(0, 3))
 
     logging.info("Verifying no error in logs")
     assert_no_errors_in_logs(remote_command_executor, scheduler)
 
 
-def _assert_scaling_works(
```
Comment (demartinofra) [inline tests/integration-tests/tests/scaling/test_scaling.py:73]: I think we should not remove the asg part but just rename it cause it was already checking against ec2 capacity in case of Slurm and we should preserve that assertion

File: cli/src/pcluster/templates/cluster_stack.py [hunk 9]
```diff
@@ -508,18 +508,6 @@ def _add_pcluster_policies_to_role(self, role_ref: str, name: str):
                     iam.PolicyStatement(
                         sid="DynamoDBList", actions=["dynamodb:ListTables"], effect=iam.Effect.ALLOW, resources=["*"]
```
Comment (demartinofra) [inline cli/src/pcluster/templates/cluster_stack.py:509]: I think you can remove this as well

File: tests/integration-tests/tests/update/test_update.py [hunk 10]
```diff
@@ -488,7 +472,7 @@ def _check_update_compute(
     cluster.config_file = str(updated_config_file)
     cluster.update()
 
-    logging.info("Sleeping for 180 seconds in case instance type properties cache is not updated yet in jobwatcher")
+    logging.info("Sleeping for 180 seconds in case instance type properties cache is not updated yet in compute config")
```
Comment (demartinofra) [inline tests/integration-tests/tests/update/test_update.py:475]: I think we can remove this and the sleep

Comment (enrico-usai) [reply tests/integration-tests/tests/update/test_update.py]: Done

File: cli/src/pcluster/templates/cluster_stack.py (hunk 9 above)
Comment (enrico-usai) [reply cli/src/pcluster/templates/cluster_stack.py]: done

File: tests/integration-tests/tests/scaling/test_scaling.py (hunk 8 above)
Comment (enrico-usai) [reply tests/integration-tests/tests/scaling/test_scaling.py:73]: Done, thanks!

**Review by demartinofra — APPROVED** [review]
